akka-sse/src/main/scala/de/heikoseeberger/akkasse/client/EventStreamClient.scala

```diff
@@ -105,17 +105,19 @@ object EventStreamClient {
   )(implicit ec: ExecutionContext, mat: Materializer): Source[A, NotUsed] = {
 
     def getAndHandleEvents = {
-      def getAndHandle(lastEventId: Option[String]) = {
+      def getAndHandle(lastEventId: Option[String])
+        : (Future[Option[ServerSentEvent]], A) = {
         import EventStreamUnmarshalling._
         val request = {
           val r = Get(uri).addHeader(Accept(`text/event-stream`))
           lastEventId.foldLeft(r)((r, i) => r.addHeader(`Last-Event-ID`(i)))
         }
-        val events = send(request).flatMap(Unmarshal(_).to[EventStream])
+        val events      = send(request).flatMap(Unmarshal(_).to[EventStream])
+        val emptySource = Future.successful(Source.empty[ServerSentEvent])
         Source
-          .fromFuture(events)
+          .fromFuture(events.fallbackTo(emptySource))
```

Comment:
So, for some reason the test suite passes even if failures are turned into empty responses.
So either we don't care about failure propagation or we have some missing test cases here.